sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -2195,4 +2195,143 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
     }
   }
 
+  /**
+   * Parse new column info from ADD COLUMN into a NewColumn.
+   */
+  override def visitNewColumn(ctx: NewColumnContext): AnyRef = withOrigin(ctx) {
+    NewColumn(
+      ctx.name.identifier.asScala.map(_.getText),
```

Comment:
Should we add a `parseQualifiedName` for this?  `ctx.<name>.identifier.asScala.map(_.getText)` for qualifiedName seems to be used in many places.

Reply:
+1

Reply:
Fixed.